enterprise/functions/src/test/java/io/crate/window/LagFunctionTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.window;
+
+import io.crate.execution.engine.window.AbstractWindowFunctionTest;
+import io.crate.metadata.ColumnIdent;
+import io.crate.module.EnterpriseFunctionsModule;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.Map;
+
+import static io.crate.analyze.WindowDefinition.CURRENT_ROW_UNBOUNDED_FOLLOWING;
+import static org.hamcrest.Matchers.contains;
+
+public class LagFunctionTest extends AbstractWindowFunctionTest {
+
+    public LagFunctionTest() {
+        super(new EnterpriseFunctionsModule());
+    }
+
+    @Test
+    public void testLagEmptyOver() throws Exception {
```

Comment:
Would be nice to have a bit more verbose test names that explain the scenarios a bit better.

"Could you re-create the test body by reading the test name"